### PR TITLE
[Welcome] Add a greeting pool for returning users

### LINKED ISSUE #478: [Welcome] Send different messages on guild rejoin
Sometimes users leave and rejoin a server. Add the ability to send a different welcome message on rejoins. This will require storing a user's ID if they leave.

### PR DESCRIPTION
**Pull request by quachtridat**
## Summary

This PR creates a separate pool containing the greeting entries for those who rejoin servers. When it happens, the returning member will get a random message pulled out from the said greeting pool.

Also this PR refactors some docstring. Multiple docstring changes are done in a separate commit, if that helps viewing Git diffs.

## Details

The cog basically records user IDs when they enter a server for the first time, and uses the list of recorded user IDs to decide which greeting pool to use to send greetings to joining members.

There are 2 pools for now:
- `DEFAULT`: for greetings sent to new members.
- `RETURNING`: for greetings sent to returning members. If this pool is empty, the `DEFAULT` pool will be used.

### Commands

Extend one extra argument to some commands to accomodate this change:
- `[p]welcomeset greetings add <name> [pool]`
- `[p]welcomeset greetings remove <name> [pool]`
  - The switch from `del dict[key]` to `dict.pop(key, default)` is because `del` may throw a `KeyError`.
- `[p]welcomeset greetings list [pool]`
  - The pool's name is shown in the embed footer.

## Linked issues
This resolves #478.

### CODE REVIEW

**Review by Injabie3 — CHANGES_REQUESTED** [review]
LGTM, one comment below. Also, in the following: https://github.com/SFUAnime/Ren/blob/3095461521a5edfa8b5dd5021b25a8d8af90ef74/cogs/welcome/welcome.py#L220-L225

Can we add the notion of pools to this group's docstring? Doesn't have to be on the first line, but on a subsequent line so that if someone just goes to the group (i.e. `[p]welcomeset greetings`) to see the commands, they'll know that the commands under that group will accept a `pool` parameter. I don't think anyone would know about pools unless someone explicitly asked for the help on `add`/`rm`/`ls`.

**Comment by quachtridat** [comment]
> LGTM, one comment below. Also, in the following:
> 
> https://github.com/SFUAnime/Ren/blob/3095461521a5edfa8b5dd5021b25a8d8af90ef74/cogs/welcome/welcome.py#L220-L225
> 
> Can we add the notion of pools to this group's docstring? Doesn't have to be on the first line, but on a subsequent line so that if someone just goes to the group (i.e. `[p]welcomeset greetings`) to see the commands, they'll know that the commands under that group will accept a `pool` parameter. I don't think anyone would know about pools unless someone explicitly asked for the help on `add`/`rm`/`ls`.

Take a look at my new commits.

File: cogs/welcome/welcome.py [hunk 1]
```diff
@@ -29,8 +30,31 @@ def __init__(self, bot: Red):
...
+            The guild to get a random greeting from.
+        pool: Optional[GreetingPools]
+            The pool to get a random greeting from.
+        """
+        if pool is None:
+            pool = GreetingPools.DEFAULT
+
+        key = KEY_GREETINGS
+        if pool == GreetingPools.RETURNING:
+            key = KEY_RETURNING_GREETINGS
```
Comment (Injabie3) [inline cogs/welcome/welcome.py]: Not sure where my previous comment went, so I'll repost.

Can we get rid of the `is None` check? Doesn't seem useful in this case: we can just compare `None` directly with the enum int:
```py
>>> from enum import IntEnum
>>> class TestClass(IntEnum):
...    DEFAULT = 0
...    TEST = 1
>>> None == TestClass.TEST
False
>>> None == TestClass.DEFAULT
False
>>> 1 == TestClass.DEFAULT
False
>>> 0 == TestClass.DEFAULT
True
```
```suggestion
        key = KEY_GREETINGS
        if pool == GreetingPools.RETURNING:
            key = KEY_RETURNING_GREETINGS
```

**Review by Injabie3 — CHANGES_REQUESTED** [review]

File: cogs/welcome/welcome.py (hunk 1 above)
Comment (quachtridat) [reply cogs/welcome/welcome.py]: Thanks. Take a look at my new commit.

**Review by Injabie3 — APPROVED** [review]
thanks lgtm

**Pull request merged**